common/src/main/java/com/tc/net/protocol/tcm/CommunicationsManagerImpl.java

```diff
@@ -175,8 +177,11 @@ public CommunicationsManagerImpl(MessageMonitor monitor, TCMessageRouter message
     this.connectionManager = connMgr;
 
     Assert.eval(healthCheckerConfig != null);
+    ReferenceQueue<Object> gc = new ReferenceQueue<>();
+    PhantomReference<Object> ref = new PhantomReference<>(this, gc);
```

Comment:
From memory phantom references are more expensive during GC than weak ones. Not that there'll be many of them... but does this need to be phantom?

Reply:
I was under the impression that phantom references where a replacement for finalization which this is basically an implementation of.